credoai/prism/prism.py

```diff
@@ -0,0 +1,49 @@
+from typing import List
+from credoai.lens import Lens
+from credoai.utils import ValidationError, global_logger
+from credoai.utils.common import NotRunError
+from credoai.prism.compare import Compare
+
+
+class Prism:
+    SUPPORTED_TASKS = {"compare": Compare}
+
+    def __init__(self, network: List[Lens], tasks: dict):
+        self.network = network
```

Comment:
The usage of this class is pretty opaque.

`Network` is a word that comes out of nowhere. This is a list of lens objects? Do we also want to support lists of evidence containers here? Seems a bit restricted in scope while also not being that transparent.

`Tasks` is very opaque. I feel like we should automatically be able to run comparators, with optional parameterization. I also feel like you _should not_ have to define reference class by string, but instead by model.